src/Proj4.jl

```diff
@@ -125,8 +130,9 @@ function transform!(src::Projection, dest::Projection, position::Array{Float64,2
     P = pointer(position)
     x = P
     y = P + sizeof(Cdouble)*npoints
-    z = ncomps < 3 ? C_NULL : P + 2*sizeof(Cdouble)*npoints
-    err = _transform!(src, dest, npoints, 1, x, y, z)
+    z = P + 2*sizeof(Cdouble)*npoints
+    err = (ncomps < 3) ? _transform!(src, dest, npoints, Int32(1), x, y) :
+          _transform!(src, dest, npoints, Int32(1), x, y, z)
```

Comment:
In the original version, the type of `z` varies between `Ptr{Float64}` and `Ptr{Void}`, depending on the value of `ncomps`.

Ideally, it'd be better to determine from the data-type of `z` which version of `transform!` to dispatch it to, rather than based on the run-time `length`, but since we've decided to be relaxed about supporting vectors (which can be arbitrary in length), we have to perform these runtime checks.